lockfile_unix.go

```diff
@@ -75,6 +71,14 @@ func (l *lockfile) lock(l_type int16) {
 	}
 	l.stateMutex.Lock()
 	if l.counter == 0 {
+		// If we're the first reference on the lock, we need to open the file again.
```

Comment:
(BTW, above and below, setting `Pid` does not make sense. This is a return-only field.)

Reply:
This sounds plausible but I seem to be unable to find information about that without having to read kernel code.

Let's defer that to a future PR.

Reply:
Really? All of fcntl(2), fcntl(3p), (info libc) say this about `l_pid`.

Reply:
I must have been temporarily blind...it's all over the place.